src/components/List/utils/ListUtils.ts

```diff
@@ -1,10 +1,15 @@
-import { MutableRefObject } from "react";
+import { MutableRefObject, useState } from "react";
 import { ListWrapperComponentStringType, ListWrapperComponentType } from "../ListConstants";
 import { ListItemComponentType } from "../../ListItem/ListItemConstants";
 
 let listIdCounter = 0;
 export const generateListId = () => `list-${listIdCounter++}`;
 
+export const useListId = (id: string) => {
+  const [listId] = useState(id || generateListId());
+  return listId;
+};
```

Comment:
Why does it using state? Can't it be a regular util function?

Reply:
Cause if it will be regular function it will run on each rerender and return a new id each time

Reply:
Cause if it will be regular function it will run on each rerender and return a new id each time